compiler/vm/vmrunner.nim

```diff
@@ -275,16 +297,23 @@ proc main*(args: seq[string]): int =
       break
     of yrkError:
       # an uncaught error occurred
-      c.config.localReport(createStackTrace(c, thread))
-      c.config.localReport(r.error)
+      c.config.localReport(createLegacyStackTrace(c, thread))
+      let location = 
+        case r.error.kind
+        of vmEvtUserError:
+          r.error.srcLoc
+        else:
+          r.error.location.map(proc (l: PrgCtr): TLineInfo = c.debug[l])
+                        .get(source(c, thread))
```

Comment:
Using `c` inside the anonymous procedure makes it part of the closure's environment (that is, the `TCtx` instance is now located on the heap instead of the stack). 

`c.debug[l]` always yields the same value as `source(c, thread)`, so you can simply use the latter directly.

Reply:
Sigh, that's what I get for programming under the weather. Pretty much all the VM events will be raised and so their location and what's in the thread will be the same. Ooph